### PR TITLE
Polish README

### PR DESCRIPTION
**Pull request by xarvh**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,149 +3,269 @@ Independence
...
 
-var monkey = require('./monkey');
+monkey = require '../lib/monkey'
 
-function momentMock() {
-  return {
-    format: function() {
-      return "Yaaap!";
+# Mock the `moment` and `myDatabase` modules, but leave the original `lodash`
+mocks = {
```
Comment (runk) [inline README.md]: I guess you can omit parens here since you decided to do so in example above

File: README.md [hunk 2]
```diff
@@ -3,149 +3,269 @@ Independence
...
+This means that if in the above example the `myDatabase` or `myService` modules require
+`moment` they will still get the original one rather than the mock.
+
 
-// Provide only moment and leave all other dependencies undefined:
-var pureMonkey = monkey.independence('isolate', {moment: momentMock});
+Rebuilding a whole tree
+-----------------------
+To recursively inject mocks from a module down to all its nested dependencies you can specify
+`'rebuild'` as first argument:
```
Comment (runk) [inline README.md]: ... you can choose `rebuild` mode?

File: README.md [hunk 3]
```diff
@@ -3,149 +3,269 @@ Independence
 [![NPM version][npm-image]][npm-url] [![Dependency Status][daviddm-url]][daviddm-image]
 [![Build Status](https://secure.travis-ci.org/Adslot/independence.png?branch=master)](http://travis-ci.org/Adslot/independence)
 
-Module dependency injection for easy and fast mocking.
+Node.js module dependency injection for easy and fast mocking.
 
 Enables a module to quickly create clones of itself that have their dependencies mocked.
 
+Independence works with plain JavaScript, but the examples are given in
```
Comment (runk) [inline README.md]: I'd wrap "Independence" into stars to make it italic: `* Independence*`

File: README.md [hunk 4]
```diff
@@ -3,149 +3,269 @@ Independence
...
-testMonkey.swing() // Works as normal
+```coffeescript
+# Provide only the `moment` module and replace all other dependencies with {}
 
+pureMonkeyClone = monkey.independence 'isolate', mocks
 
-// Override moment in the monkey module and in ALL its wrapped dependencies
-// recursively (in this case, myService and myDatabase):
-var testMonkey = monkey.independence('rebuild', {moment: momentMock});
+pureMonkeyClone.swing() # Will throw `TypeError: Object #<Object> has no method 'unique'`
```
Comment (runk) [inline README.md]: `swing` isn't defined anywhere

File: README.md [hunk 5]
```diff
@@ -3,149 +3,269 @@ Independence
...
+      clone.makeDessert (err) ->
+        expect(frozen.calledOnce).to.be true
+        done()
 ```
 
 
-Coffee & Mocha
---------------
+### Throwaway Monkeys ###
+_Monkey-patched clones don't need to be restored_
```
Comment (runk) [inline README.md]: "monkey-patched monkey" slightly confusing :smile_cat:

File: README.md [hunk 6]
```diff
@@ -3,149 +3,269 @@ Independence
...
+    'integration/fruit':
+      connect: (fruit, cb) -> cb null, connection: 'up'
+      query: fruitApiResponse ? (args..., cb) -> cb null, []
+
+
+  describe 'syncFruitDessert()', ->
+
+    it 'should handle the fruit response', (done) ->
+      clone = patient.independence baseMockFruitQuery (args..., cb) -> cb null, [fruit: 'banana']
+      clone.syncDessert done
```
Comment (runk) [inline README.md]: Does "sync" stands for "synchronous"?

**Comment by runk** [comment]
Great stuff @xarvh :100:

File: README.md (hunk 6 above)
Comment (xarvh) [inline README.md]: I was thinking about "synchronise", not that it means anything since it's only an example.
Shall I change it to "saveDessert"?

Comment (runk) [inline README.md]: Yeah, guess so. It confused me a little that you have async tests and "sync" code :)

Comment (xarvh) [inline README.md]: :+1:

Comment (camjc) [inline README.md]: just make it 'eatDessert' or whatever.

**Comment by camjc** [comment]
LGTM mate. :racehorse:

**Comment by runk** [comment]
:ok_hand:

File: README.md [hunk 7]
```diff
@@ -3,149 +3,269 @@ Independence
...
+[CoffeeScript](http://coffeescript.org/) for readability.
+
 
 When writing the module
 -----------------------
-Wrap your module in
-`"require('independence')(require, module, (function(require, module, exports) {" + yourCode + "}))"`.
+Wrap your module with:
+```
+"require('independence')(require, module, (function(require, module, exports) {" + yourCode + "}))"
```
Comment (omgaz) [inline README.md]: I definitely think the string concatenation needs an explanation, or at least a note.

File: README.md [hunk 8]
```diff
@@ -3,149 +3,267 @@ Independence
...
---------------
+### Throwaway Monkeys ###
+_Monkey-patched clones don't need to be restored_
+
+```coffeescript
+patient = require 'a/module/of/mine'
+
+describe 'a/module/of/mine', ->
+
+  describe 'peelBanana()', ->
```
Comment (omgaz) [inline README.md]: Two lines between describes?

File: README.md [hunk 9]
```diff
@@ -3,149 +3,267 @@ Independence
...
 
-`mocha --compilers coffee:independence/bin/coffee-wrap`
+### Mocks as Functions ###
+_We can reuse the same mock for different tests_
+
+```coffeescript
+patient = require 'ANOTHER/module/of/mine'
+
+
+describe 'ANOTHER/module/of/mine', ->
```
Comment (omgaz) [inline README.md]: Or remove two lines here? Consistency?

File: README.md [hunk 10]
```diff
@@ -3,149 +3,267 @@ Independence
...
 
-var testController = controller.independence('isolate', {
-    'models/user': userModelMock,
-    'controllers/user': userControllerMock
-  });
+controllerClone = controller.independence 'isolate',
+  'models/user': userModelMock
+  'controllers/user': userControllerMock
 
 ...
```
Comment (omgaz) [inline README.md]: … instead of ...?

File: README.md [hunk 11]
```diff
@@ -3,149 +3,267 @@ Independence
...
 
-function momentMock() {
-  return {
-    format: function() {
-      return "Yaaap!";
-    }
-  };
-}
 
+Rebuilding a whole tree
```
Comment (omgaz) [inline README.md]: Maybe `deep independence` or `recursive dependencies`?

**Comment by omgaz** [comment]
Merging :ramen:

**Pull request merged**